### ISSUE TITLE
Possible mistake in ElphHashes

### ISSUE THREAD

**Issue by liulizhi1996** [melifluos/subgraph-sketching#8]
I think the following code is incorrect: 

https://github.com/melifluos/subgraph-sketching/blob/3732cc75d8da216b41b972620dc545ff5aa1f6e1/src/hashing.py#L281-L282

It should be:
features[:, 7] = cards1[:, 1] - torch.sum(features[:, 0:4], dim=1) - features[:, 5]  # (2, 0)

**Comment by liulizhi1996** [comment]
By the way, I think the Eq. (3) in the original paper is also wrong. The right equation should be:
B_uv[d] = |N_d(u)| - \sum_{i=1}^{d-1} B_uv[i] - \sum_{i=1}^{d} \sum_{j=1}^{d} A_uv[i, j]
The difference between above and that in your paper is the summation of B_uv[i] where i from 1 to d-1.

**Comment by shir994** [comment]
Hi @liulizhi1996. Thanks, indeed it seems that `features[:, 0]` should not be in this version of the code.
I think you are also right about the missing summation in the equation. We will fix this along with a few other bugs in a coming PR.

**Comment by lei-liu1** [comment]
Hello,
I think there is also something wrong with Equation (2). Should it be this？

![mathpix 2023-07-25 12-35-50](https://github.com/melifluos/subgraph-sketching/assets/57094098/5bf12690-862f-482f-8118-b267db5d1d74)

If I have a misunderstanding, please let me know.
Best regards,
Lei

**Comment by liulizhi1996** [comment]
@lei-liu1 Perhaps you are right. Eqn.(2) is not correct in the original paper.
